src/components/DataSession/OperationPipeline.vue

```diff
@@ -36,6 +44,40 @@ function operationBtnColor(index) {
 	}
 }
 
+async function pollOperationCompletion(operationID) {
+	const url = await store.state.datalabApiBaseUrl + 'datasessions/' + props.data.id + '/operations/' + operationID + '/'
+
+	const response = await fetchApiCall({ url: url, method: 'GET', failCallback: handleError })
+
+	if (response && response.percent_completion !== undefined && response.status != 'COMPLETED') {
+		operationPercentages.value[operationID] = response.percent_completion * DEC_TO_PERCENT
+	} else {
+		operationPercentages.value[operationID] = COMPLETE_PERCENT
+		setTimeout(stopPollingOperation(operationID), POLL_WAIT_TIME)
+	}
+}
+
+function stopPollingOperation(operationID) {
+	clearInterval(operationPollingTimers[operationID])
+	delete operationPercentages.value[operationID]
+	delete operationPollingTimers[operationID]
+}
+
+watch(() => props.operations, () => {
+	props.operations.forEach(operation => {
+		if (!operationPollingTimers[operation.id]) {
+			operationPollingTimers[operation.id] = setInterval( () => pollOperationCompletion(operation.id), POLL_WAIT_TIME)
+		}
+	})
+}, { immediate: true })
+
+onBeforeUnmount(() => {
+	// Clean up Polling Intervals
+	Object.keys(operationPollingTimers).forEach(operationID => {
```

Comment:
I think this needs an existence check for length

Reply:
Will it crash on an empty object?

Reply:
I tested on the console with an empty object like this
```
var x = {}
Object.keys(x).forEach(console.log("yes"))
```
and it returned 1 yes so I don't think it crashes but it returns erroneous data?